### DISCUSSION TITLE
大神还是有问题，以上午脚本启动的passwall表面上是启动了，但是不能用

### DISCUSSION THREAD

**Discussion by wwwliutao125** [XIU2/CloudflareSpeedTest#353] [讨论]
还得进路由器，点保存&应用，才能使用，唉，要不然就完美了

**Comment by wwwliutao125** [comment]
每跑一次，都得进后台，重新点

**Comment by XIU2** [comment]
因为不清楚你的路由器里的 passwall 具体启动命令是什么，昨天远程协助我只是找了找系统服务，当时你也没测试。

**Comment by wwwliutao125** [comment]
想想咋办，大神，要不不完美

**Comment by wwwliutao125** [comment]
我又加了定时任务，你的脚本用行完十分钟后重启路由，虽然方法笨点，然后发现还是不行，一看passwall居然没启动

**Comment by wwwliutao125** [comment]
00 5 * * * sleep 5 && touch /etc/banner && reboot
 加了好几条类似的，定时任务。再看时居然只剩下路由默认这一条了

**Comment by wwwliutao125** [comment]
好像可以的，有些糊涂

**Comment by XIU2** [comment]
你这越说越迷糊。

你是想重新改回上次的方案？（即定时重启路由器，启动 CloudflareST，延迟启动 passwall）
那你还需要改下脚本，删除关闭、开启 passwall 的代码。

**Comment by wwwliutao125** [comment]
改进现在方案，现在这个方案是有问题的，脚本跑完后，passwall是正常启动，但在等域名解析生效后，就需要进路由器后台，重新点保存并应用

**Comment by wwwliutao125** [comment]
跑完脚本后，前几分钟可以正常使用。隔几分钟后就无法使用了，只能在路由器里点一下保存和应用，具体原因不清楚

**Comment by wwwliutao125** [comment]
我只是怀疑和动态域名解析有关，域名解析间隔，是十分钟

**Comment by XIU2** [comment]
- 一个是降低域名解析 TTL，即域名解析生存时间（多久需要更新域名解析缓存），看看能不能解决。
- 一个是每次脚本测速完，加个 hosts，把域名都指向最快 IP，而 Hosts 是立即生效的。

如果你不需要在其他设备、网络下单独使用你的代理，那么连 DDNS 的步骤都能省掉。

**Comment by wwwliutao125** [comment]
我怀疑是不是需要重启v2ray

**Comment by wwwliutao125** [comment]
路由器里的保存和应用，和你代码控制，肯定有区别

**Comment by XIU2** [comment]
但问题是我不知道相应的命令呀。

**Comment by wwwliutao125** [comment]
不是ttl问题，如果是缓存，那不应该断才对，现在是不在路由器里点保存与应用，就完全不能用

**Comment by wwwliutao125** [comment]
你能不能在脚本里，后面再加个重启v2ray命令，

**Comment by XIU2** [comment]
我不是万能的。

你用的代理工具都依托于 passwall，都由 passwall 控制。

**Comment by wwwliutao125** [comment]
一个是每次脚本测速完，加个 hosts，把域名都指向最快 IP，而 Hosts 是立即生效的。，要不，用这个办法试试

**Comment by XIU2** [comment]
那就远程协助吧，希望这是最后一次。。。

**Comment by wwwliutao125** [comment]
多谢，哈哈

**Comment by wwwliutao125** [comment]
xiangrikui
